### PR TITLE
fix(formatting): resolve trailing whitespace reversal bug in splitEdgeWhitespace

### PR DESCRIPTION
**Pull request by Awesome-RJ**
## Description
This PR fixes a bug in the `splitEdgeWhitespace` helper function where trailing whitespace characters (e.g. `" \t"`) were written in reverse order (`"\t "`). Because of this reversal, `strings.TrimSuffix` failed to match and strip the trailing whitespace, leading to whitespace duplication in the final formatted outputs.

### Changes
- Rewrote `splitEdgeWhitespace` in `formatting.go` using index pointers/rune slicing directly.
- Avoided `strings.Builder` allocations and substring searches.
- Created `formatting_test.go` to test whitespace edge cases (mixed whitespace, pre blocks, leading/trailing space).

### CODE REVIEW

**Review by PaulSonOfLars — APPROVED** [review]
good spot, thank you!

**Comment by PaulSonOfLars** [comment]
@Awesome-RJ Looks like tests aren't passing - are you sure you've defined the expected behaviour correctly?

File: formatting_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			wantCnt: "",
+			wantPos: "",
+		},
+		{
+			name:    "newlines in pre",
+			text:    "\n hello \n",
+			entType: "pre",
+			wantPre: " ",
+			wantCnt: "\n hello \n",
+			wantPos: "",
```
Comment (PaulSonOfLars) [inline formatting_test.go]: Are we sure this is the right behaviour? Why would `pre` be an empty space and not `pos`?

Comment (Awesome-RJ) [reply formatting_test.go]: Good catch! You're completely right. The code behaves correctly, but I had defined the expected behavior in my unit test incorrectly (asserting wantPre: " " instead of keeping it in wantPos). I've corrected the test case.

**Comment by PaulSonOfLars** [comment]
I've got a small edge case i remember having to deal with which i want to make sure is correctly covered by this - will need to test on my side, but otherwise LGTM :)
